core/arch/arm/kernel/thread_optee_smc.c

```diff
@@ -172,8 +170,7 @@ static uint32_t call_entry_std(struct optee_msg_arg *arg, size_t num_params,
 	if (rpc_arg)
 		thr->rpc_arg = NULL;
 
-	if (rv == OPTEE_SMC_RETURN_OK &&
-	    !(IS_ENABLED(CFG_PREALLOC_RPC_CACHE) && thread_prealloc_rpc_cache))
+	if (rv == OPTEE_SMC_RETURN_OK && !thread_prealloc_rpc_cache)
```

Comment:
With this change it looks like it might be harder for the compiler to determine that `thread_prealloc_rpc_cache` is unused with `CFG_PREALLOC_RPC_CACHE=n`.
If you think the if-statement needs to be simplified feel free to add a
``` C
bool prealloc_rpc_cache_is_eanbled()